bindings/ocaml/EverCrypt.ml

```diff
@@ -85,30 +65,46 @@ module AEAD = struct
     match UInt8.to_int (everCrypt_AEAD_create_in (alg_definition alg) st (C.ctypes_buf key)) with
     | 0 -> Success (alg, st)
     | n -> error n
-  let encrypt (alg, st) iv ad pt ct tag : unit result =
-    (* EverCrypt.AEAD.encrypt_pre *)
-    check_sizes alg iv tag;
-    assert (C.disjoint ct tag);
-    assert (C.disjoint iv ct);
-    assert (C.disjoint iv tag);
-    assert (C.disjoint pt tag);
-    assert (C.disjoint pt ad);
-    assert (C.disjoint ad ct);
-    assert (C.disjoint ad tag);
-    get_result (everCrypt_AEAD_encrypt (!@st)
-                  (C.ctypes_buf iv) (C.size_uint32 iv) (C.ctypes_buf ad) (C.size_uint32 ad)
-                  (C.ctypes_buf pt) (C.size_uint32 pt) (C.ctypes_buf ct) (C.ctypes_buf tag))
-  let decrypt (alg, st) iv ad ct tag dt : unit result =
-    (* EverCrypt.AEAD.decrypt_st *)
-    check_sizes alg iv tag;
-    assert (C.disjoint tag dt);
-    assert (C.disjoint tag ct);
-    assert (C.disjoint tag ad);
-    assert (C.disjoint ct ad);
-    assert (C.disjoint dt ad);
-    get_result (everCrypt_AEAD_decrypt (!@st)
-                  (C.ctypes_buf iv) (C.size_uint32 iv) (C.ctypes_buf ad) (C.size_uint32 ad)
-                  (C.ctypes_buf ct) (C.size_uint32 ct) (C.ctypes_buf tag) (C.ctypes_buf dt))
+
+  module Noalloc = struct
+    let encrypt ~st:(alg, st) ~iv ~ad ~pt ~ct ~tag : unit result =
```

Comment:
Don't know what's the convention here or at Nomadic Labs, but my personal discipline is to annotate the argument of all functions with their types to improve code readability :)

Reply:
Fair point, but since I've annotated the functions in the .mli (in this case `val encrypt : st:t -> iv:bytes -> ad:bytes -> pt:bytes -> ct:bytes -> tag:bytes -> unit Error.result` for example) and by extension in the docs too I thought it's maybe not worth doing it again here?

Reply:
Yeah you're probably right. And the types are all the same anyway here.

Reply:
I think it's ok because of the mli. If not for the mli, I would have advocated for annotated types.